controllers/grid/queries/form/QueryNoteForm.inc.php

```diff
@@ -132,6 +132,12 @@ function execute() {
 
 		$notificationDao = DAORegistry::getDAO('NotificationDAO');
 		$queryDao = DAORegistry::getDAO('QueryDAO');
+
+		// Always include current user to query participants
+		if (!in_array($user->getId(), $queryDao->getParticipantIds($query->getId()))) {
+			$queryDao->insertParticipant($query->getId(), $user->getId());
+		}
```

Comment:
Lovely. It's great when a cool new feature can be done in 3 lines of code.

Reply:
was just thinking that it now presumes that only editors can get that far. I mean can try answering a discussion where they are not a participant. But I figured that earlier checks in the code will prevent them from getting that far.

Reply:
I think that's ok. Others should not get that far and if they can, it may be because someone has deliberately altered the permissions (say, because they want section editors to view all discussions on a submission they are assigned to, even if not assigned to a discussion). So the assumption here is correct: add anyone who is replying but not already assigned.